test/integration/shell_int_test.go

```diff
@@ -292,27 +291,33 @@ func (suite *ShellIntegrationTestSuite) TestNestedShellNotification() {
 	cfg, err := config.New()
 	suite.Require().NoError(err)
 
+	// Setup subshell. Since our CI tests are running in bash, override with zsh on macOS.
 	var ss subshell.SubShell
 	env := []string{"ACTIVESTATE_CLI_DISABLE_RUNTIME=false"}
 	if runtime.GOOS != "darwin" || !condition.OnCI() {
 		ss = subshell.New(cfg)
 	} else {
-		os.Setenv("SHELL", "zsh") // GitHub actions runs on bash, so override with zsh
+		cfg.Set(subshell.ConfigKeyShell, "zsh")
```

Comment:
Please get rid of the subshell package in these tests. We can just make assumptions based on OS. eg. Mac = zsh, Linux = bash, Windows = cmd.

The subshell package adds too much logic to these tests imo. I'd rather keep this dead simple and hardcoded so the tests leave little to the imagination and we don't accidentally break them without realizing it when eg. updating the subshell package.

Reply:
Forgive me. I was under the impression "integration tests" were meant to test as much of the existing system in-place as possible, so breaking changes introduced in the subshell packages might be caught here.